crates/experimentation_platform/src/api/experiments/helpers.rs

```diff
@@ -189,39 +197,44 @@ pub fn validate_experiment(
 }
 
 pub fn add_variant_dimension_to_ctx(
-    context_json: &Value,
+    opt_context_json: &Option<Value>,
     variant: String,
 ) -> superposition::Result<Value> {
-    let context = context_json.as_object().ok_or(bad_argument!(
-        "Context not an object. Ensure the context provided obeys the rules of JSON logic"
-    ))?;
-
-    let mut conditions = match context.get("and") {
-        Some(conditions_json) => conditions_json
-            .as_array()
-            .ok_or(bad_argument!(
-                "Failed parsing conditions as an array. Ensure the context provided obeys the rules of JSON logic"
-            ))?
-            .clone(),
-        None => vec![context_json.clone()],
-    };
-
+    let mut updated_ctx = Map::new();
     let variant_condition = serde_json::json!({
         "in" : [
             variant,
             { "var": "variantIds" }
         ]
     });
-    conditions.push(variant_condition);
-
-    let mut updated_ctx = Map::new();
-    updated_ctx.insert(String::from("and"), serde_json::Value::Array(conditions));
 
-    match serde_json::to_value(updated_ctx) {
-        Ok(value) => Ok(value),
-        Err(_) => Err(bad_argument!(
-            "Failed to convert context to a valid JSON object. Check the request sent for correctness"
-        )),
+    if let Some(context_json) = opt_context_json {
```

Comment:
Lets do case match for `if let` conditions having else condition too.
Improves readability.

Reply:
@mahatoankitkumar I'm sorry, i didn't get it 
there is if let condition along with else for the opt_context_json

Reply:
use case match

Reply:
Sure

Reply:
Done